### PR TITLE
initial changes until splitters

### PR DESCRIPTION
**Pull request by ngupta23**
#### Describe the changes you've made
- Initial changes fir upgrade to sktime 0.7.0. 
- Fixed updated for metrics and BaseForecaster
- Now errors out at the Splitters

```python
Traceback (most recent call last):
  File "c:/Users/Nikhil/OneDrive/my_libraries/my_python_libraries/pycaret_dev/time_series_debug.py", line 41, in <module>
    exp.setup(data=y, fh=fh, fold=fold, fold_strategy="expanding", session_id=42, n_jobs=-1)
  File "c:\Users\Nikhil\OneDrive\my_libraries\my_python_libraries\pycaret_dev\pycaret\internal\pycaret_experiment\time_series_experiment.py", line 1062, in setup
    profile_kwargs=profile_kwargs,
  File "c:\Users\Nikhil\OneDrive\my_libraries\my_python_libraries\pycaret_dev\pycaret\internal\pycaret_experiment\tabular_experiment.py", line 1570, in setup
    self.fold_generator = self.get_fold_generator(fold=self.fold_param)
  File "c:\Users\Nikhil\OneDrive\my_libraries\my_python_libraries\pycaret_dev\pycaret\internal\pycaret_experiment\time_series_experiment.py", line 3245, in get_fold_generator
    start_with_window=True,
TypeError: __init__() got an unexpected keyword argument 'window_length'
```

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

Not tested yet since more changes are needed to complete the upgrade to sktime 0.7.0

## Checklist:
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: pycaret/containers/models/time_series.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 import pandas as pd
 import logging
 
-from sktime.forecasting.base._sktime import _SktimeForecaster  # type: ignore
+from sktime.forecasting.base import BaseForecaster  # type: ignore
```
Comment (TremaMiguel) [inline pycaret/containers/models/time_series.py:18]: just for curiosity, does BaseForecaster has different default methods than _SktimeForecaster ?

**Review by TremaMiguel — APPROVED** [review]

**Comment by TremaMiguel** [comment]
Everything looks good, just some import is breaking the ensemble method with the change to sktime 0.7.0, gonna take a look at it.

File: pycaret/containers/models/time_series.py (hunk 1 above)
Comment (ngupta23) [reply pycaret/containers/models/time_series.py:18]: I need to double check this, but this is what they have done in their library as well (sktime)

Comment (ngupta23) [reply pycaret/containers/models/time_series.py:18]: _OptionalForecastingHorizonMixin was also removed if I remember correctly.

**Pull request merged**